src/Core/SemanticVersion.cs

```diff
@@ -39,32 +45,49 @@ public SemanticVersion(int major, int minor, int build, string specialVersion)
         {
         }
 
+        public SemanticVersion(int major, int minor, int build, string specialVersion, string metadata)

```

Comment:
In SemVer terms, the `build` argument is called `patch`. Perhaps we should use the same terminology from the SemVer.org spec to improve readability?

According to the spec, major, minor and patch numbers are mandatory, and labels for prerelease versions and build metadata are optional version parts. Knowing the SemVer spec, it is not clear to me what the `specialVersion` and `metadata` arguments refer to.

`Major.Minor.Patch-PrereleaseLabel+BuildMetadata`